### ISSUE TITLE
Memory Leak/Growth when using: http-request redirect code 30x location %[path,map_str(redirect_uris_30x.map)] ignore-empty

### ISSUE THREAD

**Issue by f-leclere** [haproxy/haproxy#1675]
### Detailed Description of the Problem

I have HAProxy 2.5.5 in SSL termination config running in a container of an openshift POD, the backend is a wordpress instance that runs in another container on another POD.

Bug reproduction are done using local wsl2 ubuntu 20.04 with linux docker and using same docker image as on openshift but without ssl termination (could also be done on non openshift platform)

### Expected Behavior

container memory stay stable under 1Gb with normal traffic (10-20 cnx/min)

### Steps to Reproduce the Behavior

any http request even them to be redirected

### Do you have any idea what may have caused this?

Our prod team finaly point the two http-request redirect causing this.
              http-request redirect code 301 location %[path,map_str(redirect_uris_301.map)] ignore-empty
              http-request redirect code 302 location %[path,map_str(redirect_uris_302.map)] ignore-empty

redirect_uris_301.map file contain line
          /ha301check /v1/status

redirect_uris_302.map file contain line
          /ha302check /products

### Do you have an idea how to solve the issue?

no, except comment these two lines

### What is your configuration?

```haproxy
runtime: openshift 4.7 or wsl2 ubuntu 20.04 with linux docker
docker image os base: 
first os when bug detected: alpine_3.15.4
second os to exclude a musl issue: debian 11.3 build with jemalloc

1 Single HAProxy Server (1Gb of ram)
1 or 2 wordpress apache_2.4 Servers

/ $ Debian 11.3 bullseye
/ $ uname -a
prod: 
Linux prod-haproxy-57d475d767-djkhv 4.18.0-305.40.2.el8_4.x86_64 #1 SMP Tue Mar 8 14:29:54 EST 2022 x86_64 GNU/Linux
local:
Linux 18149da1b402 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 GNU/Linux

configuration is splitted in two files
backend server named "wordpress" is a localhost alias

haproxy.cfg
------------
# Global config
global
    default-path config
    pidfile /run/haproxy.pid
    daemon
    localpeer localglobalpeer
    profiling.memory on

    tune.ssl.cachesize 100000
    tune.ssl.lifetime 600
    tune.ssl.maxrecord 1460
    tune.ssl.default-dh-param 2048

    ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!aECDH:!RC4
    ssl-default-bind-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
    ssl-default-bind-options prefer-client-ciphers no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets

    ssl-default-server-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!aECDH:!RC4
    ssl-default-server-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
    ssl-default-server-options no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets

    spread-checks 5
    ca-base /etc/ssl/certs

    stats socket /run/haproxy.sock mode 666 level admin
    log stdout len 65535 local2

defaults
    mode http
    retries 3
    option redispatch
    option forwardfor if-none
    option http-server-close
    option http-keep-alive
    option http-buffer-request
    maxconn 4096
    timeout check 2000
    timeout connect 2000
    timeout client 12000
    timeout http-request 10s
    timeout http-keep-alive 5000
    timeout server 29000
    timeout queue 29s

    option dontlognull
    option http-ignore-probes
    option log-separate-errors
    option log-health-checks
    no option logasap   ### disable early logging of HTTP requests so that total transfer time is logged ###
    log-format %ci:%cp\ [%t]\ %ft:\ |[%T]|[.%ms]|%b|%s|%ST|%sslc|%sslv|%Tq/%Tw/%Tc/%Tr/%Tt|%B|%ac/%fc/%bc/%sc/%rc|%sq/%bq|ID\ %ID|cookie_c\ %CC|cookie_s\ %CS|CN\ %[ssl_fc_sni]|Header_c\ %hr|Header_s\ %hs|%{+Q}r|Upload:\ %U|Term-state:\ %ts|%Th|%Ti|%TR|%{+E}[var(txn.acl_trigger)]|
    unique-id-format %{+X}o\ %ci:%cp_%fi:%fp_%ts_%rt:%pid
    unique-id-header X-UUID

peers od-haproxy-peers
    bind *:10024
    server localglobalpeer

resolvers default-resolver
    parse-resolv-conf
    timeout resolve       20s
    timeout retry         500ms
    resolve_retries       4
    hold other           30s
    hold refused         30s
    hold nx              30s
    hold timeout         30s
    hold valid           30s
    hold obsolete        30s
    accepted_payload_size 8192 # allow larger DNS payloads

frontend probe
    log stdout len 65535 local2
    bind *:10079
    mode http
    monitor-uri /status

frontend stats
    log stdout len 65535 local1
    bind *:10080
    mode http
    http-request use-service prometheus-exporter if { path /metrics }
    stats enable
    stats uri /stats
    stats show-modules
    stats show-legends
    stats refresh 10s

frontdev.cfg
------------
frontend od-frontdev-http-frontend
    log stdout len 65535 local0
    mode    http
    option  forwardfor

    option httpclose
    tcp-request inspect-delay 11s

    acl h_xff_exists req.hdr(X-Forwarded-For) -m found

    capture request header HOST len 100
    capture request header X-Forwarded-host len 500
    capture request header X-FORWARDED-FOR len 500
    capture request header FORWARDED len 500
    capture cookie "${FRONTDEV_COOKIE_name}" len 30
    capture request header X-UUID len 100
    capture request header X-OAPI-Application-Id len 40
    capture request header X-OAPI-Application-Name len 100
    capture request header X-OAPI-User-Id len 10
    capture request header X-OAPI-Request-Id  len 40
    capture request header SM_REALM len 30
    capture request header SM_AUTHTYPE len 20
    capture request header SM_UNIVERSALID len 10
    capture request header SM_USERDN len 256
    capture request header FTUSERGIVENNAME len 100
    capture request header FTUSERSN len 100
    capture request header FTUSERMAIL len 256
    capture request header User-Agent len 220
    capture request header Referer len 200
    capture request header Pragma len 10
    capture request header Cache-Control len 100
    capture request header Connection len 10
    capture request header Content-Length len 10
    capture request header Accept-Encoding len 50
    capture request header Cookie len 1024
    capture request header Code len 200

    capture response header X-OAPI-Request-Id  len 40
    capture response header Cache-Control len 100
    capture response header content-encoding len 40
    capture response header content-type len 40
    capture response header content-length len 40
    capture response header Last-Modified len 40
    capture response header date len 40
    capture response header expires len 40
    capture response header Pragma len 10
    capture response header ETag len 50
    capture response header Via len 30
    capture response header location len 200
    capture response header Connection len 10
    capture response header Content-Encoded len 10
    capture response header Vary len 50
    capture response header Age len 10
    capture response header set-cookie len 1024

    bind *:11080

    acl h_xff_exists req.hdr(x-forwarded-for) -m found
    http-request set-header x-forwarded-for %[src] unless h_xff_exists

    http-request track-sc0 req.hdr_ip(x-forwarded-for) table od_dev_st_xff_global

    acl h_xfh_exists req.hdr(X-Forwarded-Host) -m found
    http-request set-header X-Forwarded-Host %[req.hdr(host)] unless h_xfh_exists
    acl h_xfport_exists req.hdr(X-Forwarded-Port) -m found
    http-request set-header X-Forwarded-Port %[dst_port] unless h_xfport_exists
    acl h_xfproto_exists req.hdr(X-Forwarded-Proto) -m found
    http-request set-header X-Forwarded-Proto http if !{ ssl_fc } !h_xfproto_exists
    http-request set-header X-Forwarded-Proto https if { ssl_fc } !h_xfproto_exists

    http-request redirect code 301 location %[path,map_str(redirect_uris_301.map)] ignore-empty
    http-request redirect code 302 location %[path,map_str(redirect_uris_302.map)] ignore-empty

    default_backend od-frontdev-backend

backend od-frontdev-backend
    log stdout len 65535 local1
    mode http
    compression algo gzip raw-deflate identity
    compression type text/plain text/xml text/css text/html 'text/html; charset=UTF-8' text/json text/javascript font/woff font/ttf text/json image/avif image/webp image/svg+xml image/png image/jpeg image/pjpeg image/gif image/bmp image/tiff image/x-icon image/webp application/javascript application/json application/pdf application/xhtml+xml 'application/xml;q=0.9' application/vnd.ms-powerpoint application/vnd.ms-project application/vnd.visio application/x-7z-compressed application/zip application/x-tar '*/*;q=0.8'
	
    http-request add-header X-OD-via "od-haproxy"
    http-request set-header X-Client-IP %[req.hdr_ip(X-Forwarded-For)]

    # Remove a response header
    http-after-response del-header Server
    http-after-response del-header server
    http-after-response del-header X-Powered-By

    acl be_od_dev_is_jscss path_end .js .css .woff
    acl be_od_dev_is_image path_end .png .jpg .jpeg
    http-request set-priority-class int(1) if be_od_dev_is_jscss
    http-request set-priority-class int(10) if be_od_dev_is_image
    http-request set-priority-class int(100) if !be_od_dev_is_jscss !be_od_dev_is_image

    http-response set-header Server-Timing miss,\ 0_TR_client_wait;dur=%TR,\ 1_Tw_queues;dur=%Tw,\ 2_Tc_bak_connect;dur=%Tc,\ 3_Tr_bak_response;dur=%Tr

    http-request set-var(txn.path) path
    acl is_wp_uploads_path var(txn.path) -i -m beg /wp-content/uploads/ /od-uploads/
    http-request cache-use wp_uploads if is_wp_uploads_path
    http-response cache-store wp_uploads if is_wp_uploads_path

    acl is_wp_frontdev_app_path var(txn.path) -i -m beg /wp-content/ /wp-admin /wp-includes/ /od-content/ /od-admin/ /od-plugins/ /od-muplugins/
    http-request cache-use wp_frontdev_app if is_wp_frontdev_app_path
    http-response cache-store wp_frontdev_app if is_wp_frontdev_app_path

    acl is_ope_contents_path var(txn.path) -i -m beg /ope-contents/
    http-request cache-use ope_contents if is_ope_contents_path
    http-response cache-store ope_contents if is_ope_contents_path

    balance leastconn

    option tcp-check
    tcp-check connect port 8080
    option http-pretend-keepalive
    server od-frontdev wordpress:8080 resolvers default-resolver resolve-prefer ipv4 check maxconn 500 check inter 5s fastinter 1s downinter 2s rise 2 fall 3 maxqueue 20 on-marked-down shutdown-sessions init-addr libc,last,none

######################################################################################################
cache wp_frontdev_app
    process-vary on
    max-secondary-entries 10
    total-max-size 50
    max-object-size 2097152
    max-age 87000

cache wp_uploads
    process-vary on
    max-secondary-entries 10
    total-max-size 100
    max-object-size 2097152
    max-age 87000

cache ope_contents
    process-vary on
    max-secondary-entries 10
    total-max-size 50
    max-object-size 2097152
    max-age 87000
######################################################################################################
backend od_dev_st_xff_global
   stick-table type ipv6 size 1M expire 36h store conn_rate(1m),conn_cur,gpt0,gpc0,gpc1,http_req_rate(10m),http_err_rate(10m)
```

### Output of `haproxy -vv`

```plain
Output of production haproxy -vv
/$ haproxy -vv
HAProxy version 2.5.5-384c5c5 2022/03/14 - https://haproxy.org/
Status: stable branch - will stop receiving fixes around Q1 2023.
Known bugs: http://www.haproxy.org/bugs/bugs-2.5.5.html
Running on: Linux 4.18.0-305.40.2.el8_4.x86_64 #1 SMP Tue Mar 8 14:29:54 EST 2022 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_LINUX_TPROXY=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_TFO=1 USE_PROMEX=1 USE_MEMORY_PROFILING=1
  DEBUG   = 

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL +THREAD +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX +MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=8).
Built with OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
Running on OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.36 2020-12-04
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace

------------------------------
Output of local haproxy -vv

$ haproxy -vv
HAProxy version 2.5.5-384c5c5 2022/03/14 - https://haproxy.org/
Status: stable branch - will stop receiving fixes around Q1 2023.
Known bugs: http://www.haproxy.org/bugs/bugs-2.5.5.html
Running on: Linux 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_LINUX_TPROXY=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_TFO=1 USE_PROMEX=1 USE_MEMORY_PROFILING=1
  DEBUG   = 

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL +THREAD +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX +MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
Running on OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.32 2018-09-10
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 8.3.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace
```

### Last Outputs and Backtraces

```plain
Production "show pools" without redirect lines

/$ echo "show pools"| socat stdio /run/haproxy.sock 
Dumping pools usage. Use SIGQUIT to flush them.
  - Pool cache_st (32 bytes) : 97 allocated (3104 bytes), 97 used (~89 by thread caches), needed_avg 93, 0 failures, 42 users, @0x7f2bfe092c40 [SHARED]
  - Pool hlua_flt_ct (64 bytes) : 416 allocated (26624 bytes), 416 used (~384 by thread caches), needed_avg 412, 0 failures, 48 users, @0x7f2bfe084840 [SHARED]
  - Pool vars (96 bytes) : 70 allocated (6720 bytes), 70 used (~70 by thread caches), needed_avg 66, 0 failures, 5 users, @0x7f2bfe095440 [SHARED]
  - Pool ssl_sock_ct (128 bytes) : 4021 allocated (514688 bytes), 4021 used (~4019 by thread caches), needed_avg 4017, 0 failures, 25 users, @0x7f2bfe083440 [SHARED]
  - Pool task (160 bytes) : 177 allocated (28320 bytes), 177 used (~112 by thread caches), needed_avg 173, 0 failures, 4 users, @0x7f2bfe096840 [SHARED]
  - Pool h2s (192 bytes) : 4030 allocated (773760 bytes), 4030 used (~4027 by thread caches), needed_avg 4026, 0 failures, 7 users, @0x7f2bfe085c40 [SHARED]
  - Pool tcpcheck_ru (224 bytes) : 186 allocated (41664 bytes), 186 used (~56 by thread caches), needed_avg 182, 0 failures, 17 users, @0x7f2bfe08c840 [SHARED]
  - Pool ssl_sock_cl (256 bytes) : 12 allocated (3072 bytes), 12 used (~11 by thread caches), needed_avg 8, 0 failures, 3 users, @0x7f2bfe082040 [SHARED]
  - Pool caphdr (288 bytes) : 17 allocated (4896 bytes), 17 used (~17 by thread caches), needed_avg 13, 0 failures, 6 users, @0x7f2bfe09a440 [SHARED]
  - Pool spoe_ctx (320 bytes) : 20 allocated (6400 bytes), 20 used (~20 by thread caches), needed_avg 16, 0 failures, 2 users, @0x7f2bfe08f040 [SHARED]
  - Pool resolv_reso (480 bytes) : 3 allocated (1440 bytes), 3 used (~0 by thread caches), needed_avg 1, 0 failures, 1 users, @0x7f2bfe090440 [SHARED]
  - Pool caphdr (512 bytes) : 47 allocated (24064 bytes), 47 used (~47 by thread caches), needed_avg 43, 0 failures, 9 users, @0x7f2bfe099040 [SHARED]
  - Pool resolv_answ (608 bytes) : 20 allocated (12160 bytes), 20 used (~8 by thread caches), needed_avg 16, 0 failures, 1 users, @0x7f2bfe091840 [SHARED]
  - Pool requri (1024 bytes) : 26 allocated (26624 bytes), 26 used (~26 by thread caches), needed_avg 22, 0 failures, 1 users, @0x7f2bfccd0040 [SHARED]
  - Pool caphdr (1056 bytes) : 16 allocated (16896 bytes), 16 used (~16 by thread caches), needed_avg 12, 0 failures, 4 users, @0x7f2bfe09b840 [SHARED]
  - Pool stream (1152 bytes) : 26 allocated (29952 bytes), 26 used (~25 by thread caches), needed_avg 22, 0 failures, 1 users, @0x7f2bfe08dc40 [SHARED]
  - Pool fcgi_conn (1216 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7f2bfe08b440 [SHARED]
  - Pool h2c (1312 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7f2bfe08a040 [SHARED]
  - Pool caphdr (2080 bytes) : 20 allocated (41600 bytes), 20 used (~20 by thread caches), needed_avg 16, 0 failures, 2 users, @0x7f2bfe09cc40 [SHARED]
  - Pool hpack_tbl (4096 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7f2bfccd2840 [SHARED]
  - Pool buffer (16384 bytes) : 47 allocated (770048 bytes), 47 used (~29 by thread caches), needed_avg 43, 0 failures, 1 users, @0x7f2bfccd1440 [SHARED]
  - Pool trash (16416 bytes) : 8 allocated (131328 bytes), 8 used (~7 by thread caches), needed_avg 4, 0 failures, 1 users, @0x7f2bfe097c40
  - Pool dns_msg_buf (65568 bytes) : 0 allocated (0 bytes), 0 used (~0 by thread caches), needed_avg 0, 0 failures, 1 users, @0x7f2bfe094040 [SHARED]
Total: 23 pools, 2463360 bytes allocated, 2463360 used (~2099232 by thread caches).
```

**Comment by capflam** [comment]
I confirm the issue. It is caused by `ignore-empty` option. The leak was introduced by the commit bc1223be79. Unfortunately, the 2.5.6 was released earlier today... Anyway, thanks for report !

**Comment by capflam** [comment]
a workaround is to use an ACL instead of `ignore-empty` option:
```
http-request redirect code 301 location %[path,map_str(redirect_uris_301.map)] if { path,map_str(redirect_uris_301.map) -m found }
```

**Comment by capflam** [comment]
The fix was backported to 2.5. Thanks !

**Comment by f-leclere** [comment]
Thank's Capflam for the workaround, it's ok locally.
I request to deploy it on prod too and wait for the final 2.6 or next 2.5
